src/sidebar/services/test/api-test.js

```diff
@@ -17,395 +16,283 @@ const util = require('../../../shared/test/util');
 const routes = require('./api-index.json').links;
 
 describe('sidebar.services.api', function() {
-  let $httpBackend;
-  let $q;
   let fakeAuth;
   let fakeStore;
-  let sandbox;
   let api;
 
-  before(function() {
-    angular.module('h', []).service(
-      'api',
-      proxyquire(
-        '../api',
-        util.noCallThru({
-          angular: angular,
-          '../retry-util': {
-            retryPromiseOperation: function(fn) {
-              return fn();
-            },
-          },
-        })
-      )
-    );
-  });
-
-  beforeEach(function() {
-    sandbox = sinon.sandbox.create();
+  function defaultBodyForStatus(status) {
+    if (status === 204) {
+      return '';
+    } else if (status >= 500) {
+      return '<html><body>Internal Server Error</body></html>';
+    } else {
+      return {};
+    }
+  }
+
+  /**
+   * Expect an HTTP call using `window.fetch`.
+   *
+   * @param {string} method - Expected HTTP method (lower case)
+   * @param {string} pathAndQuery - Expected part of URL after API root
+   * @param {number} status -
+   *   Expected HTTP status. If <= 0 then the call to `fetch` will reject with
```

Comment:
Is the HTTP status <= 0 notion here related in any way to how `fetch` behaves in real life or a test metaphor?

Reply:
In AngularJS it was the case that the "status" value was 0 or negative if the request failed. In the case of `fetch` it will reject with an error so there is no HTTP status to speak of. Perhaps `null` would be better.

Reply:
`null` is nice because it implies a non-value versus a specific (bad) one...